service-base/src/main/java/org/eclipse/hono/service/tenant/TenantHttpEndpoint.java

```diff
@@ -72,45 +72,56 @@ public void addRoutes(final Router router) {
         final BodyHandler bodyHandler = BodyHandler.create();
         bodyHandler.setBodyLimit(config.getMaxPayloadSize());
 
-        // ADD tenant
+        // ADD tenant with auto-generated ID
         router.post(path).handler(bodyHandler);
         router.post(path).handler(this::extractRequiredJsonPayload);
-        router.post(path).handler(this::checkPayloadForTenantId);
+        router.post(path).handler(this::extractIfMatchVersionParam);
         router.post(path).handler(this::addTenant);
 
         final String pathWithTenant = String.format("/%s/:%s", TenantConstants.TENANT_ENDPOINT, PARAM_TENANT_ID);
 
+        // ADD tenant
```

Comment:
my understanding is that you want to expose the API under base URI `/tenants`, right? If so, then you should use the value returned by *getName* instead of `TenantConstants.TENANT_ENDPOINT` when creating the routes ...